rxgroups-android/src/main/java/com/airbnb/rxgroups/LifecycleResubscription.java

```diff
@@ -62,21 +79,47 @@ private boolean shouldCheckClass(@Nullable Class<?> clazz) {
     return !qualifiedName.startsWith("android.") && !qualifiedName.startsWith("java.");
   }
 
-  private Observable<ObserverInfo> observerInfoList(Object target, Field field) {
+  private Observable<ObserverInfo> observerInfoList(final Object target, Field field) {
     final Observer<?> observer;
     try {
       observer = (Observer<?>) field.get(target);
     } catch (IllegalAccessException e) {
       throw new RuntimeException(
-          String.format("Error accessing observer %s. Make sure it's no private.", field), e);
+          String.format("Error accessing observer %s. Make sure it's not private.", field), e);
     }
 
-    return Observable.from(field.getAnnotation(AutoResubscribe.class).value())
-        .map(new Func1<String, ObserverInfo>() {
-          @Override public ObserverInfo call(String tag) {
-            return new ObserverInfo(tag, observer);
+    return Observable.just(field).flatMap(new Func1<Field, Observable<ObserverInfo>>() {
+      @Override public Observable<ObserverInfo> call(Field field) {
+        //noinspection TryWithIdenticalCatches
+        try {
+          Class<?> fieldClass = observer.getClass();
+          Method tagMethod = fieldClass.getDeclaredMethod("resubscriptionTag");
+          Object tag = tagMethod.invoke(observer);
+          Class<?> rawParameterType = Utils.getRawType(tag.getClass());
```

Comment:
I think only the reflection needs to be in the try block, once we get the Object we can be outside the block, right?